misc/matrix_distribution.md

```diff
@@ -0,0 +1,92 @@
+# Matrix Distribution
+
+The matrix (also denoted global matrix) is the matrix which is distributed using a 2D block cyclic distribution scheme.
+Therefore any of the rank stores in its own memory only a part the elements which are part of the local matrix.
+The matrix is divided in submatrices of the same size (exceptions are the submatrices in the boundaries which can be smaller) called tiles.
+
+## Global to local mapping
+
+The global matrix of size `size` (denoted by blue lines in Fig. 1) is divided in blocks of size `block_size` (red lines).
+The blocks are then distributed ciclically among the processes which are mapped in a 2D grid. In Fig. 1 there are 6 processes on a (3x2) grid.
+
+![Fig. 1 Distribution of a matrix. Each color represents a process.](figures/distribution.png)
+*Fig. 1 Distribution of a matrix. Each color represents a process.*
+
+An index is used to indicate the source rank, i.e. the rank which has the top left block of the matrix.
+For example, Fig. 2 illustrates how the first (4x4) blocks are distributed for different source rank.
+
+![Fig. 2. Mapping of blocks and processes with different source rank. On the left the source rank is {0, 0}, in the center is {2, 1} and on the right is {1, 0}.](figures/src_rank.png)
+*Fig. 2. Mapping of blocks and processes with different source rank. On the left the source rank is {0, 0}, in the center is {2, 1} and on the right is {1, 0}.*
+
+Fig. 3 shows the local matrix of size `local_size` (green lines in Fig. 3) of each of the 6 processes.
+
+![Fig. 3. Local matrix of each process for the matrix depicted in Fig 1.](figures/local.png)
+*Fig. 3. Local matrix of each process for the matrix depicted in Fig 1.*
+
+## Distribution constrains
+
+The following constrains holds:
+- Each tile is stored in the memory of a single rank.
+- Each block of the 2D block cyclic distribution is a tile.
+
+## Naming conventions
+
+Strong type 2D indices and sizes are used to reduce the possibility of index mixing errors.
+The following naming convention is used for the types of indices and sizes:
```

Comment:
I would add a sort of "generator expression" trying to make it easier to remember them:

ReferenceWhatType = {Global, Local, Tile}{Element, Tile}{Index, Size}
From right to left: **Type** of/in **What** w.r.t./of **Reference**

e.g. TileElementIndex: Index of Element w.r.t. Tile
e.g. GlobalElementSize: Size in Element(s) of Global (matrix)